### ISSUE TITLE
Why I cannot debug a code in async block in intellij

### ISSUE THREAD

**Issue by lahirug** [scala/async#136]
Anybody noticed this and found a workaround ?

**Comment by efiko12** [comment]
Yes I have the same problem, but It doesn't happen every time.

**Comment by pkolaczk** [comment]
I noticed this. Annoying as hell. Actually this is probably #1 on my priority list. This and error handling.

**Comment by worldofprasanna** [comment]
Yes. Its very annoying. We used Play framework and everything is written within async block. It happens always for me.

**Comment by retronym** [comment]
I see that the IntelliJ Scala Plugin team has just added some support for scala-async to the debugger.

https://github.com/JetBrains/intellij-scala/commit/64774debae7900a2ffa5e5d602c8a03eebbe96b4
https://github.com/JetBrains/intellij-scala/commit/ff04b9f61d247e0bce57d9d7b648a30c7d5f30ff

The bug report, [SCL-9039](https://youtrack.jetbrains.com/issue/SCL-9039), seems the be reporting the same problem you're discussing here.

As a nice bonus, @niktrop seems to have added special support in "evaluate expression" so that local variables in the user-written code are retrieved from the name-mangled fields of the state machine class. Thanks!

@lahirug @efiko12 @pkolaczk @worldofprasanna Thanks for your patience with this problem. Could I ask you to test out the [latest nightly](https://confluence.jetbrains.com/display/SCA/Scala+Plugin+Nightly) of the Scala Plugin (this requires use of IDEA 15) and report whether the situation is improved?

**Comment by niktrop** [comment]
There are some problems with availability of local variables and fields. Try this example:
```
object AsyncBugs {
  def main(args: Array[String]) {
    val future = async {
      val f1 = async { false }
      val f2 = async { 42 }
      if (await(f1))
        await(f2)
      else {
        "Hi" //breakpoint here
      }
    }

    Await.result(future, Duration.Inf)
  }
}
```
Field, generated for f2 is already null. And local variable f1 is not visible from the breakpoint location. I suppose wrong debug info for local variable table is generated.

**Comment by pkolaczk** [comment]
With the recent plugin I can put a breakpoint inside of async block and now it properly stops there. However, I noticed I can't step over await calls. Much better than before, but still needs more work.
